### PR TITLE
HDDS-9447. Redundant ACL checks in getKeyInfo for S3 use case

### PR DESCRIPTION
**Pull request by duongkame**
## What changes were proposed in this pull request?

Redundant ACL checks in getKeyInfo for S3 use case. See details in the JIRA. 

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-9447

## How was this patch tested?

Should be covered by the existing tests in CI.

### CODE REVIEW

**Comment by kerneltime** [comment]
cc @SaketaChalamchala @tanvipenumudy @swamirishi @xBis7 can you please take a look.

**Review by xBis7 — APPROVED** [review]
@duongkame Thanks for the patch. Since, we are checking the S3 volume ACLs for all bucket layouts [here](https://github.com/apache/ozone/blob/master/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneAclUtils.java#L119-L122), on the `OmMetadataReader`, it's redundant to check the volume ACLs again, when validating for a key.

Changes LGTM!

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java [hunk 1]
```diff
@@ -3515,9 +3519,18 @@ public S3VolumeContext getS3VolumeContext() throws IOException {
       }
     }
 
-    // getVolumeInfo() performs acl checks and checks volume existence.
+
+    final OmVolumeArgs volumeInfo;
+    if (internal) {
```
Comment (tanvipenumudy) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java:3524]: A minor nit: can we consider renaming `internal` to something more descriptive, like `skipChecks`?

**Review by tanvipenumudy — APPROVED** [review]
Thank you @duongkame for the patch, the changes LGTM! Added a minor nit.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java [hunk 2]
```diff
@@ -3515,9 +3519,19 @@ public S3VolumeContext getS3VolumeContext() throws IOException {
...
 
-    // getVolumeInfo() performs acl checks and checks volume existence.
+
+    final OmVolumeArgs volumeInfo;
+    if (skipChecks) {
+      // for internal usages, skip acl checks and metrics.
+      volumeInfo = volumeManager.getVolumeInfo(s3Volume);
+    } else {
+      // if external usages, getVolumeInfo() performs acl checks
+      // and update metrics.
```
Comment (smengcl) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java:3529]:
```suggestion
      // for external usages, getVolumeInfo() performs acl checks
      // and metric updates.
```

**Review by smengcl — APPROVED** [review]
Thanks @duongkame for the patch. One minor comment nit inline.

**Comment by adoroszlai** [comment]
Thanks @duongkame for the patch, @smengcl, @tanvipenumudy, @xBis7 for the review.

**Pull request merged**